.github/workflows/workflow.yml

```diff
@@ -95,46 +96,98 @@ jobs:
+      changed_env_6: ${{ steps.changed.outputs.changed_env_6 }}
+      changed_env_7: ${{ steps.changed.outputs.changed_env_7 }}
+      changed_env_8: ${{ steps.changed.outputs.changed_env_8 }}
+      changed_env_9: ${{ steps.changed.outputs.changed_env_9 }}
+      changed_env_10: ${{ steps.changed.outputs.changed_env_10 }}
+      changed_env_11: ${{ steps.changed.outputs.changed_env_11 }}
+      changed_env_12: ${{ steps.changed.outputs.changed_env_12 }}
+      changed_env_13: ${{ steps.changed.outputs.changed_env_13 }}
+      changed_env_14: ${{ steps.changed.outputs.changed_env_14 }}
+      changed_env_15: ${{ steps.changed.outputs.changed_env_15 }}
+      changed_env_16: ${{ steps.changed.outputs.changed_env_16 }}
+      changed_env_17: ${{ steps.changed.outputs.changed_env_17 }}
+      changed_env_18: ${{ steps.changed.outputs.changed_env_18 }}
+      changed_env_19: ${{ steps.changed.outputs.changed_env_19 }}
+      changed_env_20: ${{ steps.changed.outputs.changed_env_20 }}
+      changed_env_21: ${{ steps.changed.outputs.changed_env_21 }}
+      changed_env_22: ${{ steps.changed.outputs.changed_env_22 }}
+      changed_env_23: ${{ steps.changed.outputs.changed_env_23 }}
+      changed_env_24: ${{ steps.changed.outputs.changed_env_24 }}
+      changed_env_25: ${{ steps.changed.outputs.changed_env_25 }}
+      changed_env_26: ${{ steps.changed.outputs.changed_env_26 }}
+      changed_env_27: ${{ steps.changed.outputs.changed_env_27 }}
+      changed_env_28: ${{ steps.changed.outputs.changed_env_28 }}
+      changed_env_29: ${{ steps.changed.outputs.changed_env_29 }}
+      changed_env_30: ${{ steps.changed.outputs.changed_env_30 }}
+      changed_env_31: ${{ steps.changed.outputs.changed_env_31 }}
+      changed_env_32: ${{ steps.changed.outputs.changed_env_32 }}
+      changed_env_33: ${{ steps.changed.outputs.changed_env_33 }}
+      changed_env_34: ${{ steps.changed.outputs.changed_env_34 }}
+      changed_env_35: ${{ steps.changed.outputs.changed_env_35 }}
+      changed_env_36: ${{ steps.changed.outputs.changed_env_36 }}
+      changed_env_37: ${{ steps.changed.outputs.changed_env_37 }}
+      changed_env_38: ${{ steps.changed.outputs.changed_env_38 }}
+      changed_env_39: ${{ steps.changed.outputs.changed_env_39 }}
+      changed_env_40: ${{ steps.changed.outputs.changed_env_40 }}
+      changed_env_41: ${{ steps.changed.outputs.changed_env_41 }}
+      changed_env_42: ${{ steps.changed.outputs.changed_env_42 }}
+      changed_env_43: ${{ steps.changed.outputs.changed_env_43 }}
+      changed_env_44: ${{ steps.changed.outputs.changed_env_44 }}
+      changed_env_45: ${{ steps.changed.outputs.changed_env_45 }}
+      changed_env_46: ${{ steps.changed.outputs.changed_env_46 }}
+      changed_env_47: ${{ steps.changed.outputs.changed_env_47 }}
+      changed_env_48: ${{ steps.changed.outputs.changed_env_48 }}
+      changed_env_49: ${{ steps.changed.outputs.changed_env_49 }}
```

Comment:
So what the heckin heck is going on here you may wondering

We added the collect job because
* you don't have a way to refer to an individual matrixed job to grab its outputs
* matrixed jobs overwrite their outputs
* we need to know which, and if any environments actually had changes

However; you can also do some hacky stuff:
* matrixed jobs overall outputs are combined, not just the outputs of the last job
* output resolution is lazy and conditional
  * e.g. the output `changed_env_0` only exists if `steps.changed.outputs.changed_env_0` exists
* this means as long as we can map a dynamic environment name to a staticly declared output, we can output which plans produced changes, and consume that via `outputs.*` below
* this _does_ also mean there's static limit to how many environments a user can define

So overall, this gets rid of the collect job, which improves our runner consumption/costs/latency